Engine/TestPlanExecuteArgs.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+namespace OpenTap;
+
+/// <summary>
+/// Arguments struct for TestPlan.Execute
+/// </summary>
+public sealed class TestPlanExecuteArgs
+{
+    /// <summary> The result listeners used in the run. If not set, this will get the default value of ResultSettings.Current.</summary>
+    public IEnumerable<IResultListener> ResultListeners { get; set; } = ResultSettings.Current;
+
+    /// <summary> Additional Parameters for the run.</summary>
+    public IEnumerable<ResultParameter> Parameters { get; set; }
+
+    /// <summary> This is set when it is only wanted to run a subset of the steps in the test plan.</summary>
+    public ISet<ITestStep> StepsOverride { get; set; }
```

Comment:
I know we are already using the parameter "StepsOverride", but I never liked it. Can we call it something even more descriptive such as "StepsToExecute" or "SelectedTestSteps"?